reputation/reputation_test.go

```diff
@@ -0,0 +1,412 @@
+package reputation
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestCreateTable(t *testing.T) {
+	db := startDB("./TestCreateTable.db")
+
+	createStmt := `
+	CREATE table foo (id interger not null primary key, name text);
+	`
+	createTable(createStmt, db)
+
+	os.Remove("./TestCreateTable.db")
+}
+
+func TestRowInsertAndQuery(t *testing.T) {
+	db := startDB("./TestRowInsertAndQuery.db")
+
+	createStmt := `
+	CREATE table reputation (
+		name text not null primary key,
+		timestamp timestamp DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')), 
+		uptime interger,
+		audit_success interger,
+		audit_fail interger,
+		latency interger,
+		amount_of_data_stored interger,
+		false_claims interger,
+		shards_modified interger
+	);`
+
+	insertStmt := `INSERT into reputation (
+		name,
+		uptime,
+		audit_success,
+		audit_fail,
+		latency,
+		amount_of_data_stored,
+		false_claims,
+		shards_modified
+	) values (?, ?, ?, ?, ?, ?, ?, ?);`
+
+	selectStmt := `SELECT 
+		name,
+		timestamp,
+		uptime,
+		audit_success,
+		audit_fail,
+		latency,
+		amount_of_data_stored,
+		false_claims,
+		shards_modified
+	FROM reputation
+	LIMIT 2`
+
+	createTable(createStmt, db)
+
+	insertRows(db, createRandRows(100), insertStmt)
+
+	selectFromDB(db, selectStmt)
+
+	cleanUpDB(db)
+
+	os.Remove("./TestRowInsertAndQuery.db")
+}
+
+func TestFindNode(t *testing.T) {
+	db := startDB("./TestFindNode.db")
+
+	createStmt := `
+	CREATE table reputation (
+		name text not null,
+		timestamp timestamp DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')), 
+		uptime interger,
+		audit_success interger,
+		audit_fail interger,
+		latency interger,
+		amount_of_data_stored interger,
+		false_claims interger,
+		shards_modified interger,
+	PRIMARY KEY(name, timestamp)
+	);`
+
+	insertStmt := `INSERT into reputation (
+		name,
+		uptime,
+		audit_success,
+		audit_fail,
+		latency,
+		amount_of_data_stored,
+		false_claims,
+		shards_modified
+	) values (?, ?, ?, ?, ?, ?, ?, ?);`
+
+	selectStmt := `SELECT 
+		name,
+		timestamp,
+		uptime,
+		audit_success,
+		audit_fail,
+		latency,
+		amount_of_data_stored,
+		false_claims,
+		shards_modified
+	FROM reputation`
+
+	createTable(createStmt, db)
+
+	rows := []RepRow{
+		RepRow{"Alice", "", 5, 10, 5, 5, 100, 0, 0},
+		RepRow{"Bob", "", 10, 20, 0, 10, 100, 0, 0},
+		RepRow{"Carol", "", 50, 10, 5, 3, 100, 0, 0},
+		RepRow{"Dave", "", 15, 10, 0, 5, 500, 0, 0},
+		RepRow{"Eve", "", 5, 10, 5, 5, 100, 0, 1},
+	}
+
+	insertRows(db, rows, insertStmt)
+
+	bestRep := naiveReputation(db, selectStmt)
+
+	if bestRep.name != "Carol" {
+		t.Error(
+			"expected Carol got", bestRep.name,
+			"with a reputation of", bestRep.naiveRep(),
+		)
+	}
+
+	cleanUpDB(db)
+
+	os.Remove("./TestFindNode.db")
+}
+
+func TestUpdateNode(t *testing.T) {
+	db := startDB("./TestUpdateNode.db")
+
+	createStmt := `
+	CREATE table reputation (
```

Comment:
i think some of these sql statements belong in the actual package and not just in the tests?

Reply:
Do I need to make a file just for sql? Are we planning on having a schema registry or something for database schema migration?

Reply:
might make sense. one thing you could do is use a pregenerated sql creation tool like https://github.com/spacemonkeygo/dbx/